store/tikv/gcworker/gc_worker.go

```diff
@@ -425,6 +431,10 @@ func resolveLocks(ctx goctx.Context, store tikv.Storage, safePoint uint64, ident
 		default:
 		}
 
+		// we'd better keep 'req.ScanLock.StartKey' the same as 'key' to avoid
+		// the error 'key is not in region' when the region split for several pieces, or
```

Comment:
Slit and merge cannot cause `key is not in region` error, but region error? Please confirm.

Reply:
`region error` will not too.
In the code below
```
if regionErr != nil {
   ...
   continue
}
```
then we will retry and get the  right region.

Reply:
I mean you should fix this comment.

Reply:
ok, I will add the situation to the comments.